gammapy/stats/variability.py

```diff
@@ -286,6 +287,86 @@ def structure_function(flux, flux_err, time, tdelta_precision=5):
     return sf, distances
 
 
+def discrete_correlation(flux1, flux_err1, flux2, flux_err2, time1, time2, tau, axis=0):
+    """Compute the discrete correlation function for a variable source.
+
+    Parameters
+    ----------
+    flux1, flux_err1: `~astropy.units.Quantity`
+        The first set of measured fluxes and associated error.
+    flux2, flux_err2 : `~astropy.units.Quantity`
+        The second set of measured fluxes and associated error.
+    time1, time2 : `~astropy.units.Quantity`
+        The time coordinates at which the fluxes are measured.
+    tau : `~astropy.units.Quantity`
+        Size of the bins to compute the discrete correlation.
+    axis : int, optional
+        Axis along which the correlation is computed.
+        Default is 0.
+
+    Returns
+    -------
+    bincenters, discrete_correlation, discrete_correlation_err : `~numpy.ndarray`, `~astropy.units.Quantity`
+        Array of discrete time bins, discrete correlation function and associated error.
+
+    References
+    ----------
+    .. [Edelson1988] "THE DISCRETE CORRELATION FUNCTION: A NEW METHOD FOR ANALYZING
+    UNEVENLY SAMPLED VARIABILITY DATA", Edelson et al. (1988)
+    https://ui.adsabs.harvard.edu/abs/1988ApJ...333..646E/abstract
+    """
+
+    flux1 = np.rollaxis(flux1, axis, 0)
+    flux2 = np.rollaxis(flux2, axis, 0)
+
+    tau = tau.to(time1.unit)
+
+    mean1, mean2 = np.nanmean(flux1, axis=0), np.nanmean(flux2, axis=0)
+    sigma1, sigma2 = np.nanstd(flux1, axis=0), np.nanstd(flux2, axis=0)
+
+    udcf1 = (flux1 - mean1) / np.sqrt((sigma1**2 - np.nanmean(flux_err1, axis=0) ** 2))
+    udcf2 = (flux2 - mean2) / np.sqrt((sigma2**2 - np.nanmean(flux_err2, axis=0) ** 2))
+
+    udcf = np.empty(((flux1.shape[0],) + flux2.shape))
+    dist = u.Quantity(np.empty(((flux1.shape[0], flux2.shape[0]))), unit=time1.unit)
+
+    for i, x1 in enumerate(udcf1):
+        for j, x2 in enumerate(udcf2):
+            udcf[i, j, ...] = x1 * x2
+            dist[i, j] = time1[i] - time2[j]
+
+    maxfactor = np.floor(np.amax(dist) / tau).value + 1
+    minfactor = np.floor(np.amin(dist) / tau).value
+
+    bins = (
+        np.linspace(
+            minfactor, maxfactor, int(np.abs(maxfactor) + np.abs(minfactor) + 1)
+        )
+        * tau
+    )
+
+    bin_indices = np.digitize(dist, bins).flatten()
+
+    udcf = np.reshape(udcf, (udcf.shape[0] * udcf.shape[1], -1))
```

Comment:
```suggestion
    udcf = udcf.reshape(-1, udcf.shape[-1])
```

Reply:
I think this simplification should work

Reply:
Thanks, unfortunately, it does not seem to produce the same results